t/unit-tests/t-reftable-tree.c

```diff
@@ -44,17 +40,17 @@ static void test_tree(void)
 	} while (i != 1);
 
 	for (i = 1; i < ARRAY_SIZE(nodes); i++) {
-		EXPECT(values + i == nodes[i]->key);
-		EXPECT(nodes[i] ==
-		       tree_search(values + i, &root, &test_compare, 0));
+		check(values + i == nodes[i]->key);
```

Comment:
In 11ce77b5cc (strvec: add functions to replace and remove strings, 2024-05-27) I'm adding a new function `check_pointer_eq()` that would be a good fit here. That commit is part of "ps/leakfixes", which will be merged to the master branch soonish.

You may want to rebase your patch series on top of that series or wait for it to land in "master" so that you can use that function here.

Reply:
I think I'll wait for the new testing functionality to hit master. Until then, I will work on the other tests.